sktime/transformations/compose/_ytox.py

```diff
@@ -1,35 +1,41 @@
-"""Use endogeneous as exogeneous features transformer."""
+"""Use endogeneous as exogenous features transformer."""
 
 # copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
 
-__author__ = ["fkiraly"]
+__author__ = ["fkiraly", "avishkarsonni"]
 __all__ = ["YtoX"]
 
 from sktime.transformations.base import BaseTransformer
 
 
 class YtoX(BaseTransformer):
-    """Create exogeneous features which are a copy of the endogenous data.
+    """Create exogenous features from a copy of the endogenous data.
 
-    Replaces exogeneous features (``X``) by endogeneous data (``y``).
+    Replaces exogenous features (``X``) with the endogenous data (``y``), optionally
+    applying a transformer to ``y`` before using it as exogenous data.
 
     To *add* instead of *replace*, use ``FeatureUnion``.
 
     Common use cases include:
 
-    * creating exogeneous variables from transformed endogenous variables
-    * creating exogeneous data from index, if no exogeneous data is available
+    * creating exogenous variables from transformed endogenous variables
+    * creating exogenous data from the index, if no exogenous data is available
     * manual construction of reduction strategies, in combination with ``YfromX``
 
     Parameters
     ----------
     subset_index : boolean, optional, default=False
-        if True, subsets the output of ``transform`` to ``X.index``,
-        i.e., outputs ``y.loc[X.index]``
+        If True, subsets the output of ``transform`` to ``X.index``,
+        i.e., outputs ``y.loc[X.index]``.
+
+    transformer : sktime transformer, or callable optional, default=None
+        If provided, will be applied to the endogenous data (``y``)
+        before moving it to the exogenous data.
+        If general callable, must implement ``fit_transform``.
```

Comment:
I think callable means a function or similar, please correct me. Not sure what you mean by implement a specific method for a callable, can you kindly elaborate?

Reply:
callable would include only functions or objects that can execute both `fit_transform` and `inverse_transform`, providing the expected data transformation In default cases where the transformer is None. Maybe the wording is a little vague.

Reply:
I think this does not answer the question - a callable is not necessarily a class. If the arg needs to be a class - which it must be if it "has methods", the docstring should say that.

We could say, if it is a callable and not a class, we interpret it as `fit_transform` of a transformer with empty `fit`